llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -1857,6 +1857,27 @@ class integer_of_opcode<Instruction castOpcode> : GICombineRule <
 
 def integer_of_truncate : integer_of_opcode<G_TRUNC>;
 
+def anyext_undef: GICombineRule<
+   (defs root:$root),
+   (match (G_IMPLICIT_DEF $undef),
+          (G_ANYEXT $root, $undef):$Aext,
+   [{ return Helper.isUndefLegalOrBeforeLegalizer(MRI.getType(${Aext}->getOperand(0).getReg())); }]),
+   (apply [{ Helper.replaceInstWithUndef(*${Aext}); }])>;
```

Comment:
The law hasn't been written down yet and the legalizers disagree with that statement. For the moment, the legality is safer.

It would be really confusing if match returns true without any documentation or hints.

Reply:
I'll make a doc update soon if it's absolutely necessary for you to do the right thing.

Reply:
For AArch64, scalable vectors are legal in some positions, but G_IMPLICIT_DEF is not legal for scalable vectors.

Reply:
Doc update PR: #117609

It turns out we had something almost describing what we wanted anyway.

Reply:
> That's understandable since scalable vectors are not implemented at all.

That is not true. There are running GlobalISel tests with SVE every time you invoke `ninja check-llvm-codegen-aarch64`.

Reply:
Sure ok, we have a tiny amount of support. We still have 99% to go.